src/vm/jitinterface.cpp

```diff
@@ -8729,52 +8730,74 @@ static CORINFO_METHOD_HANDLE resolveVirtualMethodHelper(MethodDesc* callerMethod
     //@GENERICS: shouldn't be doing this for instantiated methods as they live elsewhere
     _ASSERTE(!pBaseMD->HasMethodInstantiation());
 
-    // Interface call devirtualization is not yet supported.
-    if (pBaseMT->IsInterface())
-    {
-        return nullptr;
-    }
-
     // Method better be virtual
     _ASSERTE(pBaseMD->IsVirtual());
 
+    MethodDesc* pDevirtMD = nullptr;
+
     TypeHandle DerivedClsHnd(derivedClass);
     MethodTable* pDerivedMT = DerivedClsHnd.GetMethodTable();
     _ASSERTE(pDerivedMT->IsRestored() && pDerivedMT->IsFullyLoaded());
-
+        
     // Can't devirtualize from __Canon.
     if (DerivedClsHnd == TypeHandle(g_pCanonMethodTableClass))
     {
         return nullptr;
     }
 
-    // The derived class should be a subclass of the the base class.
-    MethodTable* pCheckMT = pDerivedMT;
-
-    while (pCheckMT != nullptr)
+    if (pBaseMT->IsInterface())
     {
-        if (pCheckMT->HasSameTypeDefAs(pBaseMT))
+        // Interface call devirtualization.
+        //
+        // For generic interface methods we must have an ownerType to
+        // safely devirtualize.
+        if (ownerType != nullptr)
```

Comment:
Yes, that check is needed. The failing tests have instances where casts are missing and the jit ends up seeing an invalid type.
```;
; Loader\classloader\TypeGeneratorTests\TypeGeneratorTest908\Generated908\Generated908.exe
; Framework:MethodCallingTest()
; local0 is type object

IL_0113  06                ldloc.0     
IL_0114  25                dup         
IL_0115  6f 28 00 00 06    callvirt     0x6000028
```
What is the best way to do this?